operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/external/AbstractCachingDependentResource.java

```diff
@@ -3,29 +3,20 @@
 import java.util.Optional;
 
 import io.fabric8.kubernetes.api.model.HasMetadata;
-import io.javaoperatorsdk.operator.api.reconciler.dependent.EventSourceProvider;
-import io.javaoperatorsdk.operator.processing.dependent.AbstractDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.AbstractEventSourceHolderDependentResource;
 import io.javaoperatorsdk.operator.processing.event.ExternalResourceCachingEventSource;
-import io.javaoperatorsdk.operator.processing.event.source.EventSource;
 
 public abstract class AbstractCachingDependentResource<R, P extends HasMetadata>
-    extends AbstractDependentResource<R, P>
-    implements EventSourceProvider<P> {
-
-  protected ExternalResourceCachingEventSource<R, P> eventSource;
+    extends
+    AbstractEventSourceHolderDependentResource<R, P, ExternalResourceCachingEventSource<R, P>> {
   private final Class<R> resourceType;
 
   protected AbstractCachingDependentResource(Class<R> resourceType) {
     this.resourceType = resourceType;
   }
 
   public Optional<R> fetchResource(P primaryResource) {
-    return eventSource.getAssociated(primaryResource);
```

Comment:
I know that you did not changed this, but probably the featch resource should be abstract. The idea is that it does not read from the cache.

Reply:
Can we do this in a separate PR?